relay-server/src/actors/envelopes.rs

```diff
@@ -765,6 +768,69 @@ impl EnvelopeProcessor {
         });
     }
 
+    /// Validates and extracts client reports.
+    ///
+    /// At the moment client reports are primarily used to transfer outcomes from
+    /// client SDKs.  The outcomes are removed here and sent directly to the outcomes
+    /// system.
+    fn process_client_reports(&self, state: &mut ProcessEnvelopeState) {
+        let mut timestamp = None;
+        let mut discarded_events = BTreeMap::new();
+        let received = state.envelope_context.received_at;
+
+        let clock_drift_processor = ClockDriftProcessor::new(state.envelope.sent_at(), received)
+            .at_least(MINIMUM_CLOCK_DRIFT);
+
+        // we're going through all client reports but we're effectively just merging
+        // them into the first one.
+        state.envelope.retain_items(|item| {
+            if item.ty() != ItemType::ClientReport {
+                return true;
+            };
+            match ClientReport::parse(&item.payload()) {
+                Ok(report) => {
+                    for (reason, category, quantity) in report.discarded_events.into_iter() {
+                        if reason.len() > 200 {
```

Comment:
why not truncate the reason?

Reply:
No point, reasons are already whitelisted explicitly in snuba so truncating them would just mean bad reasons are dropped post kafka. No win here.